src/app/model/Pawn.ts

```diff
@@ -0,0 +1,70 @@
+import { Board } from './Board';
+import { BOARDSIZE } from './Constances';
+import { Piece } from './Piece';
+import { PieceMovement } from './PieceMovement';
+import { Column, columns, Row, Side, Square } from './Types';
+
+export class Pawn extends Piece implements PieceMovement {
+  private nextRowDifference: number = 1;
+
+  constructor(id: string, side: Side) {
+    super(id, side);
+    if (this.side !== 'WHITE') this.nextRowDifference = -1;
+    console.log(this.nextRowDifference);
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [] as Square[];
+    movesToGo = movesToGo.concat(this.goAhead(position, board), this.goDoubleAhead(position, board), this.goDiagonalAhead(position, board));
+    return movesToGo;
+  }
+
+  private goAhead(position: Square, board: Board): Square[] {
```

Comment:
Bardzo ładnie ponazywałeś te metody :)
Od razu wiedziałem o co chodzi. Choć nie wiem czy lepiej nie będzie zamiast go napisać moves.
np. movesAhead. Bo to go wskazuje na jakąś czynność - komendę - wykonanie zmiany, a my tutaj tylko odczytujemy. 
Ale to taka bardziej sugesta :)

Reply:
W skoczku wykorzystam to!